### ISSUE TITLE
Browser Websockets are not properly supported

### ISSUE THREAD

**Issue by Snazzah** [abalabahaha/eris#1000]
If eris is ran in a browser then the [browser's websocket is used in shards](https://github.com/abalabahaha/eris/blob/master/lib/gateway/Shard.js#L15), however the shard still uses the websocket as if it was still using the `ws` module. This causes this to error while connecting showing this error from [here](https://github.com/abalabahaha/eris/blob/6d4668cd134d0af6b28a63f9ba3fa2790e28954d/lib/gateway/Shard.js#L1880):
```
SyntaxError: Failed to construct 'WebSocket': The subprotocol '[object Object]' is invalid.
```
The second argument is actually used for [something else](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/WebSocket) in browsers, but there are many more changes needed if planned to use the browser's websocket.

Not sure if there is browser-friendly drop-in replacement, but eris in-browser won't work without some slight rewrite of `Shard.js`.

**Comment by knht** [comment]
Eris has no support for connecting to the gateway in browsers. Running Eris on the client also exposes your token for interfacing with Discord. It's highly inadvisable and will probably never be supported.

**Comment by Snazzah** [comment]
If connecting to the gateway is not intended, then there should be proper handling with that instead of using the browser's websocket.


> Running Eris on the client also exposes your token for interfacing with Discord.

I'm also not sure what you meant by this, if you care to elaborate a bit further. Like what "client" are you talking about for instance.

**Comment by jtsshieh** [comment]
If you run Eris in a browser with the user inputting a token, then it doesn't expose your token... Something like what lightcord does.

**Comment by abalabahaha** [comment]
There are a couple edge cases where people want a Discord client in the browser for whatever reason... Hey why not?

Riya is incorrect, we did have an initial push for browser support at some point. But it's been a while and that's decayed since few people seem to care anyway (it's been almost 3 years since that second param was added with 440e8f8, and this is I think the first report I've seen about it).

Feel free to PR fixes that don't change existing behavior (WS listeners are currently Node-style, and I think some imports also die in browser), but I don't think this will get worked on for a while.
